### PR TITLE
Change deprecated v1beta1 CRD to v1

### PR DESCRIPTION
**Pull request by orfeas0**
As per helm3's warning:
`apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition`

### CODE REVIEW

**Comment by orfeas0** [comment]
@googlebot I signed it!

**Review by liyinan926 — COMMENTED** [review]
LGTM. Can we bump the chart verson?

**Comment by liyinan926** [comment]
@orfeas0 could you please bump up the chart version in this PR? Thanks!

**Comment by orfeas0** [comment]
> @orfeas0 could you please bump up the chart version in this PR? Thanks!

Bumped version, but not appVersion. Is this ok?

File: charts/spark-operator-chart/Chart.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v2
 name: spark-operator
 description: A Helm chart for Spark on Kubernetes operator
-version: 1.1.0
+version: 1.1.1
```
Comment (liyinan926) [inline charts/spark-operator-chart/Chart.yaml:4]: Looks like this needs to be `1.1.2`.

**Comment by liyinan926** [comment]
Can you resolve the conflict?

**Comment by liyinan926** [comment]
Looks like the chart integration test failed with the following message:

```
Error: failed to install CRD crds/sparkoperator.k8s.io_scheduledsparkapplications.yaml: CustomResourceDefinition.apiextensions.k8s.io "scheduledsparkapplications.sparkoperator.k8s.io" is invalid: [spec.versions[0].schema.openAPIV3Schema: Required value: schemas are required, metadata.annotations[api-approved.kubernetes.io]: Required value: protected groups must have approval annotation "api-approved.kubernetes.io", see https://github.com/kubernetes/enhancements/pull/1111]
```

**Comment by orfeas0** [comment]
I don't get the test fail, is it because I'm missing permissions?
What's the process?

**Comment by sharkymcdongles** [comment]
> I don't get the test fail, is it because I'm missing permissions?
> What's the process?

@orfeas0 @liyinan926 

Seems you need to have a PR in kubernetes GitHub then once you have approval you add this annotation:

metadata:
  annotations:
    "api-approved.kubernetes.io": "https://github.com/kubernetes/kubernetes/pull/PRNUMBER"

I think this is only due to the fact the sparkoperator team used a protected internal URL spec for the CRD spec with:

sparkoperator.k8s.io

If it was just:

sparkoperator.spark.io

then you'd need no such annotation.

So I suppose best idea is to submit a PR or an issue on kubernetes repo and ask them how to proceed. I'll be rooting for you because I need this, but I am not part of the project so not my place to make such a request externally.

**Comment by zevisert** [comment]
I think this is superceded by https://github.com/kubeflow/spark-operator/pull/1298 and can be closed now

**Pull request closed without merging**